app/submitter/submitter.py

```diff
@@ -141,3 +144,32 @@ def send_message(self, message, tx_id, questionnaire_id, case_id=None):
         finally:
             if connection:
                 self._disconnect(connection)
+
+
+class GCSFeedbackSubmitter:
+    def __init__(self, bucket_name):
+        client = storage.Client()
+        self.bucket = client.get_bucket(bucket_name)
+
+    def upload(self, metadata, payload):
+        payload.update(metadata)
+        blob = self.bucket.blob(str(uuid4()))
+        blob.metadata = metadata
+        blob.upload_from_string(
+            json.dumps(payload).encode("utf8"), content_type="application/json"
```

Comment:
Just remembered that since you are serialising the payload directly you need to update the payload with metadata before hand.
```py
payload.update(metadata)
```